### ISSUE TITLE
dpub: Endnote references

### ISSUE THREAD

**Issue by jcsteh** [w3c/aria#450]
As I understand it, it's sometimes desirable to reference an endnote. However, the `doc-noteref` role seems specific to footnotes:

> A reference to a footnote, typically appearing as a superscripted number or symbol in the main body of text.

1. Has it been determined that endnote references are not necessary?
2. If not, is `doc-noteref` intended for use with endnotes?
3. If so, the spec needs to clarify this.
4. `doc-noteref` could be used for both footnotes and endnotes. The problem with this is that footnotes and endnotes can't be distinguished without traversing to the referenced note. Also, as I understand it, footnote and endnote references are usually distinguished visually, so I'd argue there should be an equivalent distinction in ARIA.
5. Alternatively, a new role should be introduced for endnote references.

**Comment by TzviyaSiegman** [comment]
Proposed language change to definition of doc-noteref: A reference to a footnote or endnote, typically appearing as a superscripted number or symbol in the main body of text.
It is true that endnote and footnote references are often visually distinguished, but there is no standardized method to this. Sighted users also have to follow the links. I am fine with distiniguishing between endnoteref and footnoteref, but there are other kinds as well. What if there is a sidenote. Is the author supposed to pick the best option? Others are concerned that we might be creating too many roles.
Comments from @mattgarrish and @cookiecrook appreciated.
(In an earlier version we had included only footnote not endnote. This is probably a remnant).

**Comment by mattgarrish** [comment]
I'm fine with your proposed change.

As I recall, we tried to merge endnote with footnote originally, so that's why noteref only referenced footnote. We then added endnote back because of the different way that notes are included (endnotes inherit from listitem because they're grouped at the end; footnotes from section because they occur individually near their references). We were told that putting footnote on list items would remove their list semantics. It looks like the noteref definition just didn't get updated accordingly.

I'm not sure there's much benefit in adding another link type for this case; we'd have preferred to have done without the formatting distinction, and it's not terribly relevant to the nature of the note itself. More important is the backlink to return to the reference location.

**Comment by jcsteh** [comment]
Despite there being no standardised way to visually distinguish footnote and endnote references, the distinction is almost certainly consistent within the publication. So, once the sighted reader learns which type is which, it will be immediately obvious henceforth. In contrast, it might not be possible to distinguish these note references without visual information; the user would have to follow the link every time.

While you might argue that it isn't relevant to the nature of the note, the problem is that without visual information, the note references might be ambiguous. For example, let's say there is both a reference to footnote 1 and a reference to endnote 1 on the same page. With visual information, you will be able to distinguish these; you will know that they're not referring to the same note. In contrast, without visual information, you will very probably understand them to be referring to the same note, which could be very confusing at best, misleading at worst.

**Comment by mattgarrish** [comment]
I can't say that I've seen a work that uses the same numbering system when it has both footnotes and endnotes. It runs counter to publishing tradition, in my experience.

When a publisher mixes notes within the same publication, they traditionally distinguish them by the character reference, not visually (e.g., endnotes are numerically sequenced and footnotes use symbols or alpha characters).

Does that match your experience @TzviyaSiegman ?

**Comment by cookiecrook** [comment]
@TzviyaSiegman requested I comment but I’m not certain I could do so completely without a link to an example publication that used both endnotes+footnotes. Is the change just adding "or endnote" to the prose? 

That said, I’m generally opposed to adding new API unless there is a strong case for it, so if @MattGarrish thinks publishers will consistently distinguish via counter style, I prefer Tzviya’s prose change over adding a new endnote role.

**Comment by mattgarrish** [comment]
I don't have a link to a digital publication, but I found this image while searching: https://www.flickr.com/photos/chappysmom/3375614835/

At the end of the last paragraph on the page, there is the superscripted number 37 (endnote) followed by a dagger (footnote). That's the traditional print way of distinguishing the two. I haven't seen the convention changed for digital, but Tzviya no doubt sees more of these than I do.

It sounds very confusing, even for sighted readers, to have the same numbering system for both and have to rely on styling to differentiate. There's no stopping someone from doing it, I suppose, but the use of different referents has evolved to avoid this kind of ambiguity.

**Comment by TzviyaSiegman** [comment]
As @mattgarrish noted, standard practice is to use different symbols for footnotes and endnotes. This is not just tradition but is codified in most style manuals. See for example Chicago Manual of Style, which recommends numbered endnote references and asterisk, dagger, etc for footnotes (http://www.chicagomanualofstyle.org/16/ch14/ch14_sec044.html?sessionId=4671e1ca-6057-4b78-b9a9-305728723ad7).

**Comment by mattgarrish** [comment]
To veer off slightly from the discussion so far, is our understanding correct that roles don't adapt behaviourally to their context?

As I mentioned above, we added both footnote and endnote because endnotes are typically a list at the end of a section or work, while footnotes typically occur alone near their referent. It wasn't because of major differences in the type of noting.

To have only one role, it appeared that a publisher would have to put an element with a list role around each footnote to inherit from the listitem role, or endnotes could not be marked up as a list to avoid changing the meaning of the list items if inheriting from section.

In other words, could there be a single "footnote" role that inherits from listitem and section, and the context of the element on which it's used would correctly influence how it is treated? Or did we get this right by separating the two, even though it leads to a bit of layout terminology filtering in?

**Comment by cookiecrook** [comment]
@mattgarrish wrote:
> is our understanding correct that roles don't adapt behaviourally to their context?

There is some precedent in the browsers to modify certain elements on a contextual basis.

1. HTML had a proposed change to H1 that would make it act like an H2 or H3 in nested sections. 
2. All browsers do some heuristic double-checking of tables, to make sure they are not exposing junk layout tables.
3. ARIA has a few things like this: for example, listbox can be a standalone control, or part of a composite combobox.

**Comment by mattgarrish** [comment]
We discussed on the call today and the consensus was to keep both rather than make implementation more cumbersome. There's enough historical precedent for footnotes and endnotes that trying to merge them will probably just lead to push-back from the publishing community. But thanks for the clarification, as it will help moving forward when it comes to other such potential overlaps.

The decision was also to leave noteref as a single semantic and just add the clarification Tzviya proposed above. Given the tradition of using different referent systems, sighted readers are at no more of an advantage when it comes to identifying the type of note being linked to. The DTBook format has also only used noteref for referencing and it hasn't been a source of complaint.
